src/Functions/FunctionsConversion.h

```diff
@@ -1854,7 +1883,7 @@ class FunctionConvert : public IFunction
             // toDateTime64(value, scale : Integer[, timezone: String])
             || std::is_same_v<ToDataType, DataTypeDateTime64>)
         {
-            optional_args.push_back({"timezone", &isString<IDataType>, &isColumnConst, "const String"});
+            optional_args.push_back({"timezone", &isString<IDataType>, nullptr, "String"});
```

Comment:
Maybe I miss something but is this PR not supposed to make the timezone argument non-const for `toString()`, but not for date-time-like destination types like`toDate32()`, `toDateTime()`? Looking at 1875-1884, this isn't the case.

EDIT: The other cases seem covered by l. 1950ff, can you confirm?

Reply:
Confirm, toString() case covered on line 1950